### PR TITLE
Bulk pricing merge to develop

### LINKED ISSUE #10158: [master ticket]: Introduce bulk pricing to Spartacus to inform users about discounts when buying in bulk.
**Is your feature request related to a problem? Please describe.**
In Spartacus, selected items are sold at a discounted price when bought in bulk. Currently, B2B users are not aware of these different discounts when browsing for items in Spartacus as this information is hidden. Informing the user about discounts bulk prices will encourage them to buy more and save at the same time.

**Describe the solution you'd like**

To inform users about bulk prices, modify and introduce new features to Spartacus to display this information. This feature will appear on 3 distinct pages in Spartacus: Product Details page, Product Search page and User Cart page. 

**Feature LIb** 

- [x] Create feature lib for bulk pricing in Spartacus. (#10343 )

**Sample Data**

- [x]  Make proper changes to bulk pricing sample data.

**On product details page:**  (#10162)

- [x] Display a breakdown table of the different bulk prices when buying in bulk. The table should indicate the bulk quantity range (min and max) and its corresponding price per item.

- [ ] Develop unit tests for all classes used by product details. (Bulk pricing).

**On product search page:** 

(#10163 is for range, create a new ticket for "bulk pricing available")

- [ ] Display "bulk pricing available" for products that contain bulk prices when on the product search result page.

(was "a price range" but this isn't available)

Out of scope: **On cart page:**  (#10165)

- [ ] Display a message to the user letting them know whether bulk prices are available or bulk prices as been applied to their order.

Don't forget E2E tests.

**Additional context**
For more information please visit:  https://cxwiki.sap.com/display/spar/Feature%3A+B2B+Bulk+Pricing

### PR DESCRIPTION
**Pull request by ren-tham**
Closes #10158

### CODE REVIEW

File: cypress.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (dunqan) [inline cypress.json]: 👀

File: projects/core/src/product/model/product-scope.ts [hunk 2]
```diff
@@ -3,4 +3,5 @@ export enum ProductScope {
   DETAILS = 'details',
   ATTRIBUTES = 'attributes',
   VARIANTS = 'variants',
+  BULK_PRICES = 'bulkPrices',
```
Comment (dunqan) [inline projects/core/src/product/model/product-scope.ts]: Is is possible to move this one also to the library and use type augmentation?

File: projects/storefrontapp/src/environments/b2b/b2b.feature.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  bulkPricingTranslations,
+} from '@spartacus/product/bulk-pricing/assets';
+
+import { OrderApprovalRootModule } from '@spartacus/organization/order-approval/root';
+import { B2bStorefrontModule } from '@spartacus/setup';
+import { FeatureEnvironment } from '../models/feature.model';
+
+import { BulkPricingRootModule } from '@spartacus/product/bulk-pricing/root';
+
+export const b2bFeature: FeatureEnvironment = {
```
Comment (dunqan) [inline projects/storefrontapp/src/environments/b2b/b2b.feature.ts]: 👀 
lookls like a left over?

File: projects/storefrontapp/src/environments/environment.ts [hunk 4]
```diff
@@ -8,10 +8,10 @@ import { Environment } from './models/environment.model';
 export const environment: Environment = {
   production: false,
   occBaseUrl:
-    buildProcess.env.CX_BASE_URL ??
-    'https://spartacus-devci7672.eastus.cloudapp.azure.com:9002',
+    'https://spartacus-dev2.eastus.cloudapp.azure.com:9002' ??
```
Comment (dunqan) [inline projects/storefrontapp/src/environments/environment.ts]: 👀

File: projects/storefrontapp-e2e-cypress/cypress.ci.b2b.json [hunk 5]
```diff
@@ -10,7 +10,7 @@
   "env": {
     "CLIENT_ID": "mobile_android",
     "CLIENT_SECRET": "secret",
-    "API_URL": "https://spartacus-devci7672.eastus.cloudapp.azure.com:9002",
+    "API_URL": "https://spartacus-dev2.eastus.cloudapp.azure.com:9002",
```
Comment (dunqan) [inline projects/storefrontapp-e2e-cypress/cypress.ci.b2b.json]: 👀

File: feature-libs/product/bulk-pricing/index.ts [hunk 6]
```diff
@@ -0,0 +1,3 @@
+export * from './bulk-pricing.module';
```
Comment (dunqan) [inline feature-libs/product/bulk-pricing/index.ts:1]: I'm not convinced here yet, I think we might want to create separate entry points for core, occ, and components insstead of grouping them here. Will discuss it and let you know.

File: cypress.json (hunk 1 above)
Comment (ren-tham) [reply cypress.json]: Removed file.

File: projects/storefrontapp/src/environments/environment.ts (hunk 4 above)
Comment (ren-tham) [reply projects/storefrontapp/src/environments/environment.ts]: restored env file.

File: projects/storefrontapp-e2e-cypress/cypress.ci.b2b.json (hunk 5 above)
Comment (ren-tham) [reply projects/storefrontapp-e2e-cypress/cypress.ci.b2b.json]: restored env file.

File: projects/core/src/product/model/product-scope.ts (hunk 2 above)
Comment (ren-tham) [reply projects/core/src/product/model/product-scope.ts]: Augmented enum through feature lib.

File: feature-libs/product/bulk-pricing/index.ts (hunk 6 above)
Comment (mszczepaniak) [reply feature-libs/product/bulk-pricing/index.ts:1]: Separate entry points added

File: projects/storefrontapp/src/environments/environment.ts [hunk 7]
```diff
@@ -18,6 +18,6 @@ export const environment: Environment = {
   occBaseUrl: buildProcess.env.CX_BASE_URL ?? 'https://20.83.184.244:9002',
   occApiPrefix: '/occ/v2/',
   cds: buildProcess.env.CX_CDS ?? false,
-  b2b: buildProcess.env.CX_B2B ?? false,
+  b2b: buildProcess.env.CX_B2B ?? true,
```
Comment (dunqan) [inline projects/storefrontapp/src/environments/environment.ts]: 👀

Comment (mszczepaniak) [reply projects/storefrontapp/src/environments/environment.ts]: leftover, fixed

File: projects/storefrontapp/src/environments/b2b/b2b.feature.ts (hunk 3 above)
Comment (mszczepaniak) [reply projects/storefrontapp/src/environments/b2b/b2b.feature.ts]: yup, removed the file

File: projects/storefrontapp-e2e-cypress/cypress/integration/b2b/regression/bulk-pricing/b2b-bulk-pricing.e2e-spec.ts [hunk 8]
```diff
@@ -0,0 +1,24 @@
+import * as bulkPricingActions from '../../../../helpers/b2b/b2b-bulk-pricing';
+
+describe('B2B - Bulk Pricing', () => {
+  before(() => {
+    cy.window().then((win) => win.sessionStorage.clear());
+  });
+
+  describe('Check bulk pricing table', () => {
+    it('should render pricing table for products that contain bulk prices', () => {
+      bulkPricingActions.visitProductWithBulkPrices();
```
Comment (dunqan) [inline projects/storefrontapp-e2e-cypress/cypress/integration/b2b/regression/bulk-pricing/b2b-bulk-pricing.e2e-spec.ts]: Usining helpers here makes those tests very hard to reason about (and fix, if needed) as it's hiding the actual flow under some abstraction. I would advice to inline those helpers here.

File: projects/storefrontapp-e2e-cypress/cypress/integration/b2b/regression/bulk-pricing/b2b-bulk-pricing.e2e-spec.ts [hunk 9]
```diff
@@ -0,0 +1,24 @@
+import * as bulkPricingActions from '../../../../helpers/b2b/b2b-bulk-pricing';
+
+describe('B2B - Bulk Pricing', () => {
```
Comment (dunqan) [inline projects/storefrontapp-e2e-cypress/cypress/integration/b2b/regression/bulk-pricing/b2b-bulk-pricing.e2e-spec.ts]: Can we also enable mobile test as was demoed on last sprint demo?

File: feature-libs/product/_index.scss [hunk 10]
```diff
@@ -0,0 +1,14 @@
+// we require a few bootstrap files for the CSS in this code
+@import '~bootstrap/scss/functions';
+@import '~bootstrap/scss/variables';
+@import '~bootstrap/scss/_mixins';
+
+@mixin visible-focus {
```
Comment (dunqan) [inline feature-libs/product/_index.scss]: Do we need this? And whole bootstrap dependency?

File: feature-libs/product/bulk-pricing/components/bulk-pricing-table/bulk-pricing-table.component.ts [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+import { RoutingService } from '@spartacus/core';
+import { BulkPrice } from '@spartacus/product/bulk-pricing/core';
+import { Observable } from 'rxjs';
+import { switchMap } from 'rxjs/operators';
+
+@Component({
+  selector: 'cx-bulk-pricing-table',
+  templateUrl: './bulk-pricing-table.component.html',
+})
+export class BulkPricingTableComponent implements OnInit {
```
Comment (dunqan) [inline feature-libs/product/bulk-pricing/components/bulk-pricing-table/bulk-pricing-table.component.ts:12]: This class should be exported into public api

**Review by marlass — COMMENTED** [review]
in feature-libs/product/.release-it.json instead of `ng build product --prod` you should use the `yarn build:product` script instead, so also the schematics are build correctly

**Comment by mszczepaniak** [comment]
feature-libs/product/.release-it.json adjusted

File: feature-libs/product/_index.scss (hunk 10 above)
Comment (mszczepaniak) [reply feature-libs/product/_index.scss]: No, removed

File: feature-libs/product/bulk-pricing/components/bulk-pricing-table/bulk-pricing-table.component.ts (hunk 11 above)
Comment (mszczepaniak) [reply feature-libs/product/bulk-pricing/components/bulk-pricing-table/bulk-pricing-table.component.ts:12]: done

File: projects/storefrontapp-e2e-cypress/cypress/integration/b2b/regression/bulk-pricing/b2b-bulk-pricing.e2e-spec.ts (hunk 9 above)
Comment (mszczepaniak) [reply projects/storefrontapp-e2e-cypress/cypress/integration/b2b/regression/bulk-pricing/b2b-bulk-pricing.e2e-spec.ts]: done

File: projects/storefrontapp-e2e-cypress/cypress/integration/b2b/regression/bulk-pricing/b2b-bulk-pricing.e2e-spec.ts (hunk 8 above)
Comment (mszczepaniak) [reply projects/storefrontapp-e2e-cypress/cypress/integration/b2b/regression/bulk-pricing/b2b-bulk-pricing.e2e-spec.ts]: done

**Review by dunqan — DISMISSED** [review]

**Review by dunqan — APPROVED** [review]

**Pull request merged**